pylib/spinnaker/yaml_util.py

```diff
@@ -131,6 +131,44 @@ def __resolve_value(self, value, saw, original):
   def replace(self, text):
     return self.__resolve_value(text, [], text)
 
+
+  def __get_flat_keys(self, container):
+    if not isinstance(container, dict):
+      return []
+
+    flat_keys = []
+    for key,value in container.items():
+      subkeys = self.__get_flat_keys(value)
+      if subkeys == []:
+        flat_keys.append(key)
+      else:
+        flat_keys.extend([key + '.' + subkey for subkey in subkeys])
+    return flat_keys
+
```

Comment:
How about:

  def __get_flat_keys(self, container):
    flat_keys = []
    for key,value in container.items():
      if isinstance(value, dict):
        flat_keys.extend([key '.' subkey for subkey in self.__get_flat_keys(value)])
      else:
        flat_keys.append(key)
    return flat_keys